doc/src/build/fullnode.md

```diff
@@ -288,3 +288,119 @@ If you followed the instructions for [Building from Source](#building-from-sourc
+  # mostly not needed to be changed and default is good for most use cases
+  max-transactions-in-batch: 1000
+```
+## Transaction Pruning
+
+Transaction pruning is another type of pruning that removes old transactions and effects from the db. 
+Checkpoints which are created periodically contain transaction and their effects. 
+Transaction pruning works in the background by tailing checkpoint stream as well.
+Full nodes and validators can enable transaction pruning by adding can enable transaction pruning
+by adding `num_epochs_to_retain_for_checkpoints` to their `authority-store-pruning-config` config:
+
+```yaml
+authority-store-pruning-config:
+  num-latest-epoch-dbs-to-retain: 3
+  epoch-db-pruning-period-secs: 3600
+  num-epochs-to-retain: 0
+  max-checkpoints-in-batch: 10
+  max-transactions-in-batch: 1000
+  # Number of latest epochs before which transaction and effects could be pruned.
+  # When this is N (where N >= 2), pruner can prune transaction and effects from 
+  # checkpoints upto current - N epoch. Transaction and effects from current and
+  # previous epoch are never pruned. N = 2 is a recommended setting for validators
+  # and fullnodes not serving rpc requests.
+  num_epochs_to_retain_for_checkpoints: 2
+  # Ensures that individual SST files periodically go through the compaction process.
+  # This helps reclaim disk space and avoid fragmentation issues
+  periodic-compaction-threshold-days: 1
+```
+
+## Archival Fallback
+
+Once various full nodes start pruning historical transactions and effects, it may not be always possible for
+their peers to catch up transaction and effects via state synchronization. And for this a node needs to fallback
+downloading this data from a remote aws s3 bucket. This archive has all transaction and effects data since 
+genesis and lags behind the latest network checkpoint by ten minutes. Enabling this is very highly recommended
+for all nodes. A sui node can enable this automatic archive fallback by adding the following config:
+
+```yaml
+state-archive-read-config:
+  - object-store-config:
+      object-store: "S3"
+      # Use mysten-testnet-archives for testnet 
+      # Use mysten-mainnet-archives for mainnet
+      bucket: "mysten-<testnet|mainnet>-archives"
+      # Use your aws account access key id
+      aws-access-key-id: "<AWS_ACCESS_KEY_ID>"
+      # Use your aws account secret access key
+      aws-secret-access-key: "<AWS_SECRET_ACCESS_KEY>"
+      aws-region: "us-west-2"
+      object-store-connection-limit: 20
+    # How many objects to read ahead when catching up  
+    concurrency: 5
+    # Whether to prune local state based on latest checkpoint in archive.
+    # This should stay false for most use cases
+    use-for-pruning-watermark: false
+```
+
+## Setting up your own archival fallback
+
+It is also possible to set up the archival fallback to a different (personal) s3 bucket. For this a fullnode
```

Comment:
You can also set up the archival fallback to use your own S3 bucket. To do so, you must configure your Full node to archive transactions and effects. You can enable this by adding the following config to your config file.